lib/ConversionProvider.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Pandoc;
+
+use OCP\AppFramework\Services\IAppConfig;
+use OCP\Files\Conversion\ConversionMimeProvider;
+use OCP\Files\Conversion\IConversionProvider;
+use OCP\Files\File;
+use OCP\IConfig;
+use OCP\IL10N;
+use RuntimeException;
+
+class ConversionProvider implements IConversionProvider {
+	private ?array $types = null;
+
+	public function __construct(
+		private IL10N $l10n,
+		private IConfig $config,
+		private IAppConfig $appConfig,
+	) {
+	}
+
+	private function getPDFEngine(): string {
+		if (method_exists($this->appConfig, 'getValueString')) {
```

Comment:
I dropped support for anything below Nextcloud 29 recently. Not sure when `getValueString()` was introduced, but maybe this check is obsolete by now? :thinking:

Reply:
Yes 🎉